File: Ex12.java
package Ex12;

class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public String getNome() {
        return nome;
    }

    public double getPreco() {
        return preco;
    }

    public Produto somar(Produto outroProduto) {
        double precoTotal = this.preco + outroProduto.getPreco();
        return new Produto("Combo: " + this.nome + " + " + outroProduto.getNome(), precoTotal);
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto A", 50.0);
        Produto produto2 = new Produto("Produto B", 30.0);

        Produto produtoCombinado = produto1.somar(produto2);

        System.out.println("Produto combinado: " + produtoCombinado.getNome());
        System.out.println("Preço total: R$ " + produtoCombinado.getPreco());
    }
}

File: PythonEx12/Ex12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outro):
        if isinstance(outro, Produto):
            return Produto(f"{self.nome} & {outro.nome}", self.preco + outro.preco)
        return NotImplemented

    def __str__(self):
        return f"Produto: {self.nome}, Preço: R${self.preco:.2f}"

produto1 = Produto("Produto A", 50.00)
produto2 = Produto("Produto B", 30.00)

produto_soma = produto1 + produto2

print(produto_soma)
